### PR TITLE
Migrate GAT model

### LINKED ISSUE #228: Migrate GAT model to mtenn
The `GAT` model currently lives in `asapdiscovery-ml` while the `SchNet` and `e3nn` models live in the [`mtenn`](https://github.com/choderalab/mtenn) package. This issue corresponds to and resolves choderalab/mtenn#21.

### PR DESCRIPTION
**Pull request by kaminow**
Move the GAT code from `asapdiscovery.ml.models` to the [`mtenn`](https://github.com/choderalab/mtenn) package. Requires choderalab/mtenn#22 to be merged first. Resolves #228.

## Todos
  - [x] Update to use `mtenn.GAT` model
  - [x] Get rid of current `GAT` model
  - [x] Update inference to use `mtenn` model
  - [x] Update `asapdiscovery.ml.utils.load_weights` for backwards compatibility

### CODE REVIEW

**Comment by hmacdope** [comment]
New weights are uploaded and have been added in #232. That will likely go in first.

File: asapdiscovery-ml/asapdiscovery/ml/utils.py [hunk 1]
```diff
@@ -244,104 +247,101 @@ def build_model(
...
+    try:
+        pred_readout = (
+            config["pred_r"].lower() if "pred_r" in config else pred_r.lower()
+        )
+        if pred_readout == "pic50":
+            pred_readout = mtenn.model.PIC50Readout()
+        elif pred_readout == "none":
             pred_readout = None
+        else:
+            raise ValueError(f"Uknown value for -pred_r: {pred_readout}")
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/utils.py:283]:
```suggestion
            raise ValueError(f"Unknown value for -pred_r: {pred_readout} must be one of [pic50, none]")

```

File: asapdiscovery-ml/asapdiscovery/ml/utils.py [hunk 2]
```diff
@@ -244,104 +247,101 @@ def build_model(
...
+    try:
+        comb_readout = (
+            config["comb_r"].lower() if "comb_r" in config else comb_r.lower()
+        )
+        if comb_readout == "pic50":
+            comb_readout = mtenn.model.PIC50Readout()
+        elif comb_readout == "none":
             comb_readout = None
+        else:
+            raise ValueError(f"Uknown value for -comb_r: {comb_readout}")
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/utils.py:297]:
```suggestion
            raise ValueError(f"Unknown value for -comb_r: {comb_readout} must be one of [pic50, none]")
```

File: asapdiscovery-ml/asapdiscovery/ml/utils.py [hunk 3]
```diff
@@ -244,104 +247,101 @@ def build_model(
...
+            model_params = e3nn_params
+        elif os.path.isfile(e3nn_params):
+            model_params = pkl.load(open(e3nn_params, "rb"))
+        else:
+            raise ValueError(
+                "Must provide an appropriate value for e3nn_params "
+                f"(received {e3nn_params})"
+            )
+        model = build_model_e3nn(100, *model_params[1:], config)
+        get_model = partial(
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/utils.py:327]: Nice use of `partial`, love it ;)

File: asapdiscovery-ml/asapdiscovery/ml/utils.py [hunk 4]
```diff
@@ -966,6 +967,12 @@ def load_weights(model, wts_fn, check_compatibility=False):
     except RuntimeError:
         wts_dict = torch.load(wts_fn, map_location="cpu")
 
+    # Backwards compatibility for old GAT models
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/utils.py:970]: Ah thanks for this.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Few tiny changes but looks good.

File: asapdiscovery-ml/asapdiscovery/ml/utils.py (hunk 3 above)
Comment (kaminow) [reply asapdiscovery-ml/asapdiscovery/ml/utils.py:336]: haha thanks, I actually didn't know about this functionality until I saw you use it somewhere, but partially applying functions is one of my favorite things from functional programming, so expect to see a lot more of it :)

**Review by hmacdope — APPROVED** [review]
Great work @kaminow!

**Pull request merged**